cmd/registry/cmd/root.go

```diff
@@ -30,20 +32,32 @@ import (
 	"github.com/apigee/registry/cmd/registry/cmd/search"
 	"github.com/apigee/registry/cmd/registry/cmd/upload"
 	"github.com/apigee/registry/cmd/registry/cmd/vocabulary"
+	"github.com/google/uuid"
 	"github.com/spf13/cobra"
 )
 
 func Command(ctx context.Context) *cobra.Command {
+	var taskID string
 	var cmd = &cobra.Command{
 		Use:   "registry",
 		Short: "A simple and eclectic utility for working with the API Registry",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			// Initialize global default logger with unique process identifier.
+			if len(taskID) == 0 {
+				taskID = fmt.Sprintf("[ %.8s ] ", uuid.New())
+			}
+			logger := &log.Logger{
+				Level:   log.DebugLevel,
+				Handler: text.Default,
+			}
+			log.Log = logger.WithField("uid", taskID)
```

Comment:
Can we set this in the program's main function, or is there some issue with that approach I'm not noticing?

That seems like a more appropriate spot to set the global logger variable if all else is equal. Although we don't have any reason to call this `Command` function multiple times right now, if we do in the future then setting global state here will cause strange behavior.

Reply:
This is done here because we are setting the `uid` through a rootCmd flag `log_id`. The flags are parsed at runtime ([in the execute func](https://github.com/spf13/cobra/blob/de187e874d1ca382320088f8f6d76333408e5c2e/command.go#L769)). They are not available in `main.go` when we create just the command instance `cmd := cmd.Command(ctx)`

Hence, I moved the logic to `PersistentPreRun` when we have the `log_id` value available.